### ISSUE TITLE
DeepSSM: TL-Net Joint stage log columns don't match train_log.csv header

### ISSUE THREAD

**Issue by akenmorris** [SCIInstitute/ShapeWorks#2618]
The header written to `train_log.csv` applies to the AE and T-Flank rows but not the Joint rows, so the Joint block is silently misread.

In `Python/DeepSSMUtilsPackage/DeepSSMUtils/trainer.py`:

```python
# :464  header
["Training_Stage", "Epoch", "LR", "Train_Err", "Train_Rel_Err", "Val_Err", "Val_Rel_Err", "Sec"]
# :536  AE       -> train_ae_err,     train_rel_ae_err, val_ae_err,       val_rel_ae_err   (matches)
# :646  T-Flank  -> train_tf_err,     train_rel_tf_err, val_tf_err,       val_rel_tf_err   (matches)
# :768  Joint    -> train_rel_ae_err, val_rel_ae_err,   train_rel_tf_err, val_rel_tf_err   (does not match)
```

Columns 5 and 6 swap train for validation, and the Joint rows contain no absolute errors at all, so columns 4 and 6 change units mid-file. A user reading the file sees the validation latent error under the `Val_Rel_Err` label but the validation AE error under `Train_Rel_Err`, which makes the Joint stage look like it has stalled.

Fix: log the Joint stage as `train_ae_err, train_rel_ae_err, val_ae_err, val_rel_ae_err` plus separate columns for the T-flank terms, or emit a per-stage header.

Related, same file: `scheduler.step()` is only called in the Joint loop (`:786`), not in the AE (`:468-548`) or T-Flank (`:576-658`) loops, so the logged `LR` column is the base learning rate for those stages regardless of the `decay_lr` setting.
